File: uiptv-shared/src/main/java/com/uiptv/shared/sync/RemoteSyncRequest.java
package com.uiptv.shared.sync;

public record RemoteSyncRequest(RemoteSyncDirection direction,
                                String verificationCode,
                                String requesterName,
                                RemoteSyncOptions options) {
    public RemoteSyncRequest {
        if (direction == null) {
            throw new IllegalArgumentException("direction is required");
        }
        verificationCode = verificationCode == null ? "" : verificationCode;
        requesterName = requesterName == null ? "" : requesterName;
        options = options == null ? new RemoteSyncOptions(false, false) : options;
    }
}

File: server/src/main/kotlin/com/uiptv/service/remotesync/RemoteSyncRequest.kt
package com.uiptv.service.remotesync

data class RemoteSyncRequest(
    val direction: RemoteSyncDirection,
    val verificationCode: String,
    val requesterName: String,
    val options: RemoteSyncOptions
) {
    fun direction(): RemoteSyncDirection = direction
    fun verificationCode(): String = verificationCode
    fun requesterName(): String = requesterName
    fun options(): RemoteSyncOptions = options
}
